File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/Tag.java
package quintype.com.templatecollectionwithrx.models.story;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

/**
 * An object instance representing Story Tag
 *
 * @author Imran imran@quintype.com
 * @author Madhu madhu@quintype.com
 */
public class Tag implements Parcelable {

    @SerializedName("name")
    private String name;
    @SerializedName("id")
    private String id;

    /**
     * Key for tag
     *
     * @return String key
     */
    public String key() {
        return "tag#" + name.hashCode();
    }

    /**
     * Name of tag
     *
     * @return String name
     */
    public String name() {
        return name;
    }

    /**
     * Id for tag
     *
     * @return String id
     */
    public String id() {
        return id;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.name);
        dest.writeString(this.id);
    }

    public Tag() {
    }

    public Tag(String name) {
        this.name = name;
    }

    protected Tag(Parcel in) {
        this.name = in.readString();
        this.id = in.readString();
    }

    public static final Creator<Tag> CREATOR = new Creator<Tag>() {
        public Tag createFromParcel(Parcel source) {
            return new Tag(source);
        }

        public Tag[] newArray(int size) {
            return new Tag[size];
        }
    };
}

File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/Tag.kt
package quintype.com.templatecollectionwithrx.models.story

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

/**
 * Created TemplateCollectionWithRx by rakshith on 9/4/18.
 */

class Tag : Parcelable {

    @SerializedName("name")
    private var name: String? = null
    @SerializedName("id")
    private val id: String

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(this.name)
        dest.writeString(this.id)
    }

    protected constructor(parcel: Parcel) {
        this.name = parcel.readString()
        this.id = parcel.readString()
    }

    companion object CREATOR : Parcelable.Creator<Tag> {
        override fun createFromParcel(source: Parcel): Tag {
            return Tag(source)
        }

        override fun newArray(size: Int): Array<Tag?> {
            return arrayOfNulls(size)
        }
    }
}